Gradle error finding subproject dependency version

### Describe the bug

I have a top-level `build.gradle` that specifies the version of all dependencies for the subprojects. When upgrading to 2.1, the version for a dependency in a subproject can no longer be found when a classifier is specified. See the reproducer section for details.

### Expected behavior

Successful build

### Actual behavior

```
Caused by: org.gradle.api.artifacts.ResolveException: Could not resolve all dependencies for configuration ':service3:quarkusDependency'.
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.getFailure(DefaultLenientConfiguration.java:148)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration.rethrowFailure(DefaultLenientConfiguration.java:162)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultResolvedConfiguration.rethrowFailure(DefaultResolvedConfiguration.java:45)
        at org.gradle.api.internal.artifacts.ivyservice.DefaultResolvedConfiguration.getResolvedArtifacts(DefaultResolvedConfiguration.java:83)
        at org.gradle.api.internal.artifacts.ivyservice.ErrorHandlingConfigurationResolver$ErrorHandlingResolvedConfiguration.getResolvedArtifacts(ErrorHandlingConfigurationResolver.java:337)
        at io.quarkus.gradle.dependency.ConditionalDependenciesEnabler.declareConditionalDependencies(ConditionalDependenciesEnabler.java:44)
        at io.quarkus.gradle.QuarkusPlugin.afterEvaluate(QuarkusPlugin.java:282)
...
Caused by: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find com.google.inject:guice:.
Required by:
    project :service3
```

### How to Reproduce?

top level `build.gradle`:
```
dependencies {
  implementation 'com.google.inject:guice:3.0'
}
```

subproject `build.gradle`:
```
dependencies {
  implementation 'com.google.inject:guice' // works

  testImplementation 'com.google.inject:guice::no_aop' // previously worked
  testImplementation group: 'com.google.inject', name: 'guice', classifier: 'no_aop' // previously worked
}
```

### Output of `uname -a` or `ver`

Ubuntu 20.04

### Output of `java -version`

OpenJDK 11.0.11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.1.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.1.1

### Additional information

_No response_

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/DependencyUtils.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/DependencyUtils.java
@@ -38,7 +38,7 @@ public class DependencyUtils {
         configurationCopy.getDependencies().addAll(boms);
 
         for (Configuration toDuplicate : toDuplicates) {
-            for (Dependency dependency : toDuplicate.getDependencies()) {
+            for (Dependency dependency : toDuplicate.getAllDependencies()) {
                 if (includedBuild(project, dependency.getName()) != null) {
                     continue;
                 }

--- a/integration-tests/gradle/src/main/resources/configuration-inheritance-project/build.gradle
+++ b/integration-tests/gradle/src/main/resources/configuration-inheritance-project/build.gradle
@@ -1,0 +1,30 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+group = 'com.quarkus.demo'
+version = '1.0'
+
+repositories {
+    mavenCentral()
+    if (System.properties.containsKey('maven.repo.local')) {
+        maven {
+            url System.properties.get('maven.repo.local')
+        }
+    } else {
+        mavenLocal()
+    }
+}
+
+dependencies {
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+
+    implementation 'io.quarkus:quarkus-resteasy'
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured'
+
+    implementation 'org.apache.commons:commons-collections4:4.4'
+    testImplementation 'org.apache.commons:commons-collections4::tests'
+}
+

--- a/integration-tests/gradle/src/main/resources/configuration-inheritance-project/gradle.properties
+++ b/integration-tests/gradle/src/main/resources/configuration-inheritance-project/gradle.properties
@@ -1,0 +1,2 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus

--- a/integration-tests/gradle/src/main/resources/configuration-inheritance-project/settings.gradle
+++ b/integration-tests/gradle/src/main/resources/configuration-inheritance-project/settings.gradle
@@ -1,0 +1,21 @@
+pluginManagement {
+    repositories {
+        if (System.properties.containsKey('maven.repo.local')) {
+            maven {
+                url System.properties.get('maven.repo.local')
+            }
+        } else {
+            mavenLocal()
+        }
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    //noinspection GroovyAssignabilityCheck
+    plugins {
+        id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+
+rootProject.name = 'quarkus-configuration-inheritance'
+
+

--- a/integration-tests/gradle/src/main/resources/configuration-inheritance-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
+++ b/integration-tests/gradle/src/main/resources/configuration-inheritance-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
@@ -1,0 +1,17 @@
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class HelloResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+}

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/DependencyResolutionTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/DependencyResolutionTest.java
@@ -1,0 +1,21 @@
+package io.quarkus.gradle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import org.junit.jupiter.api.Test;
+
+public class DependencyResolutionTest extends QuarkusGradleWrapperTestBase {
+    @Test
+    public void shouldResolveDependencyVersionFromSuperConfigurationProject()
+            throws IOException, URISyntaxException, InterruptedException {
+        final File projectDir = getProjectDir("configuration-inheritance-project");
+
+        final BuildResult result = runGradleWrapper(projectDir, "clean", "quarkusBuild");
+
+        assertThat(result.getTasks().get(":quarkusBuild")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
+    }
+}

